lib/http_proxy_override.dart

```diff
@@ -14,22 +14,69 @@ Future<String?> _getProxyPort() async {
 }
 
 class HttpProxyOverride extends HttpOverrides {
-  late final String? host;
-  late final String? port;
+  /// The host part of the proxy address.
+  final String? host;
 
-  HttpProxyOverride._(this.host, this.port);
+  /// The port part of the proxy address.
+  final String? port;
 
-  static Future<HttpProxyOverride> createHttpProxy() async {
-    return HttpProxyOverride._(await _getProxyHost(), await _getProxyPort());
+  /// Configures whether a secure connection to a host should be allowed with
+  /// a server certificate that cannot be authenticated by any of the trusted
+  /// root certificates.
+  final bool ignoreBadCertificates;
+
+  final SecurityContext securityContext;
+
+  HttpProxyOverride._(
+      this.host, this.port, this.ignoreBadCertificates, this.securityContext);
+
+  /// Create an instance of [HttpProxyOverride].
+  ///
+  /// Reads the configured proxy host and port from the underlying platform
+  /// and configures the [HttpClient] to use the proxy. The proxy settings are
+  /// read once at creation time.
+  ///
+  /// [ignoreBadCertificates] configures whether a secure connection to a host
+  /// should be allowed with a server certificate that cannot be authenticated
+  /// by any of our trusted root certificates.
+  /// For example, this can be useful when using debugging proxies like Charles
+  /// or mitmproxy during development.
+  /// **Do not enable this in production unless you are 100% sure.** Setting
+  /// this enables MITM attacks.
+  /// Default: `false`.
+  ///
+  /// With [securityContext] a [SecurityContext] can be provided that is used to
+  /// construct the [HttpClient]. This can be useful to provide a
+  /// [SecurityContext] that is configured with certificates that a proxy
+  /// server requires.
+  ///
+  /// Supported platforms to read proxy settings from are **iOS** and
+  /// **Android**.
+  static Future<HttpProxyOverride> create(
```

Comment:
I think since this library aims to use for debugging purposes
1. the default value of `ignoreBadCertificates ` should be true, if the user actually need to use it in production, they should set it to false
2. it's better to make the return type to nullable and return `null` if `ignoreBadCertificates`  is not true, so we do not "override" anything if `ignoreBadCertificates` is false

e.g.

```
static Future<HttpProxyOverride?> create(
      {bool ignoreBadCertificates = true,
      SecurityContext? securityContext}) async {
    if (!ignoreBadCertificates) return null;
    return HttpProxyOverride._(
        await _getProxyHost(),
        await _getProxyPort(),
        ignoreBadCertificates,
        securityContext ?? SecurityContext.defaultContext);
  }
```